src/mon/Monitor.cc

```diff
@@ -5337,6 +5337,27 @@ void Monitor::count_metadata(const string& field, Formatter *f)
   f->close_section();
 }
 
+bool Monitor::check_daemon_versions(map<int,std::pair<int, string>> &error){
+  int i = 0;
+  bool warn = false;
+  int start = 0;
+  string first_daemon = "Hello";
+  for (auto& p : mon_metadata) {
+    auto q = p.second.find("ceph_version");
+    string current_daemon = q->second;
+    if (start == 0){
+      first_daemon = current_daemon;
+      start = 1;
+    }else if ((current_daemon != first_daemon) && (start != 0)) {
```

Comment:
I don't see you comparing the version anywhere, just the daemon string name.  So in running this code it should say that all daemons except the first is incorrect.

Reply:
first_daemon isa variable that stores the ceph version of the first daemon and current_daemon is variable that stores the ceph version of the current daemon.